test/functional/shared.js

```diff
@@ -164,6 +164,7 @@ function withClient(client, callback) {
  *
  * @param {string|Array|Function} commands commands to filter for
  * @param {object} [options] options to pass on to configuration.newClient
+ * @param {Function | object} [options.connectionString] connection string options
```

Comment:
Added connection string arg with ability to use configuration like in the case for `configuration.writeConcernMax()`. I didn't end up using this and the tests passed without the specific client options, can add those options or remove this new argument for `withMonitoredClient`